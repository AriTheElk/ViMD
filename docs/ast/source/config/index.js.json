{
  "type": "Program",
  "body": [
    {
      "type": "ExportDefaultDeclaration",
      "declaration": {
        "type": "ObjectExpression",
        "properties": [
          {
            "type": "Property",
            "key": {
              "type": "Identifier",
              "name": "theme",
              "range": [
                21,
                26
              ],
              "loc": {
                "start": {
                  "line": 2,
                  "column": 4
                },
                "end": {
                  "line": 2,
                  "column": 9
                }
              }
            },
            "value": {
              "type": "Literal",
              "value": "cobalt",
              "raw": "'cobalt'",
              "range": [
                28,
                36
              ],
              "loc": {
                "start": {
                  "line": 2,
                  "column": 11
                },
                "end": {
                  "line": 2,
                  "column": 19
                }
              }
            },
            "kind": "init",
            "method": false,
            "shorthand": false,
            "computed": false,
            "range": [
              21,
              36
            ],
            "loc": {
              "start": {
                "line": 2,
                "column": 4
              },
              "end": {
                "line": 2,
                "column": 19
              }
            }
          }
        ],
        "range": [
          15,
          38
        ],
        "loc": {
          "start": {
            "line": 1,
            "column": 15
          },
          "end": {
            "line": 3,
            "column": 1
          }
        },
        "leadingComments": [],
        "trailingComments": []
      },
      "range": [
        0,
        39
      ],
      "loc": {
        "start": {
          "line": 1,
          "column": 0
        },
        "end": {
          "line": 3,
          "column": 2
        }
      }
    }
  ],
  "sourceType": "module",
  "range": [
    0,
    39
  ],
  "loc": {
    "start": {
      "line": 1,
      "column": 0
    },
    "end": {
      "line": 3,
      "column": 2
    }
  },
  "comments": []
}