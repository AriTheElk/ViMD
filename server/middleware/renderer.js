import React from "react";
import ReactDOMServer from "react-dom/server";
import { ServerStyleSheet, StyleSheetManager } from "styled-components";

// import our main App component
import App from "../../src/App";

const path = require("path");
const fs = require("fs");

export default (req, res, next) => {
  // point to the html file created by CRA's build tool
  const filePath = path.resolve(__dirname, "..", "..", "build", "index.html");

  fs.readFile(filePath, "utf8", (err, htmlData) => {
    if (err) {
      console.error("err", err);
      return res.status(404).end();
    }

    const sheet = new ServerStyleSheet();

    // render the app as a string
    const html = ReactDOMServer.renderToString(
      <StyleSheetManager sheet={sheet.instance}>
        <App />
      </StyleSheetManager>
    );

    const styleTags = sheet.getStyleTags();

    // inject the rendered app into our html and send it
    return res.send(
      htmlData.replace(
        '<div id="root"></div>',
        `${styleTags}<div id="root">${html}</div>`
      )
    );
  });
};
