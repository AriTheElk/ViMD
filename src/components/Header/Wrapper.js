import styled from 'styled-components';
import { Row } from 'hedron';

/**
 * Define styles for the header wrapper
 */
export default styled(Row)`
  background: ${(props) => props.theme.header.background};
  line-height: 34px;
  height: ${(props) => props.theme.header.height};
  max-height: ${(props) => props.theme.header.height};
`;
