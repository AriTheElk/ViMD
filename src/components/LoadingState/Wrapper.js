import styled from "styled-components";

const Wrapper = styled.div`
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  background: ${props => props.theme.header_background};

  div {
    display: inline-block;
    position: relative;
    width: 64px;
    height: 64px;
  }
  div span {
    position: absolute;
    border: 4px solid #fff;
    opacity: 1;
    border-radius: 50%;
    animation: lds-ripple 1s cubic-bezier(0, 0.2, 0.8, 1) infinite;
  }
  div span:nth-child(2) {
    animation-delay: -0.5s;
  }

  @keyframes lds-ripple {
    0% {
      top: 28px;
      left: 28px;
      width: 0;
      height: 0;
      opacity: 1;
    }
    100% {
      top: -1px;
      left: -1px;
      width: 58px;
      height: 58px;
      opacity: 0;
    }
  }
`;

export default Wrapper;
