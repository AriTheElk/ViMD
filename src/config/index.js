export default {
    theme: 'tomorrow_night',
};
