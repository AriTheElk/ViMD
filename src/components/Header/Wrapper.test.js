import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import styled, { ThemeProvider } from 'styled-components';

import Wrapper from './Wrapper';
import config from '../../config';

const theme = require(`../../styles/themes/${config.theme}`).default;


it('renders wrapper component', () => {
  const wrapper = renderer.create(
    <ThemeProvider theme={theme}>
      <Wrapper>My content</Wrapper>
    </ThemeProvider>
  );
  let tree = wrapper.toJSON();
  expect(tree).toMatchSnapshot();
});
