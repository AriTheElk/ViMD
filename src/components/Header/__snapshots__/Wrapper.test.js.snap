exports[`test renders wrapper component 1`] = `
<section
  className="kjaKCP fNeyiY">
  My content
</section>
`;
